resources/exes/arm-bounds-check/config.json

```diff
@@ -0,0 +1,84 @@
+{
+  "max-tries": 10,
+  "wp-params": {
+    "func": "sub_10468",
+    "postcond": "(assert
+(let ((base (bvadd init_R0_mod (bvmul init_R1_mod #x00000004))))
+  (implies
+    (and (= init_mem_orig init_mem_mod)
+         (bvugt init_R0_orig init_SP_orig)
+         (= init_R0_orig init_R0_mod)
+         (= init_R1_orig init_R1_mod))
+    (ite (or (bvslt init_R1_mod #x00000000)
+             (bvsge init_R1_mod #x00000003))
+      (= R0_mod #xffffffff)
+      (and (= R0_mod R0_orig)
+           (= R0_mod
+              (concat
+                (select init_mem_mod (bvadd base #x00000003))
+                (select init_mem_mod (bvadd base #x00000002))
+                (select init_mem_mod (bvadd base #x00000001))
+                (select init_mem_mod base))))))))"
+  },
+  "patch-space": [
+    {"offset": "0x510",
+     "size": "512"
+    }
+  ],
+  "patches": [
+    {"patch-name": "bounds-check",
+     "patch-code": "int *a, i, retval;
+                    if (i < 3 && i >= 0) {
+                      retval = a[i];
+                    } else {
+                      retval = -1;
+                    }",
+     "patch-point": "0x10474",
+     "patch-size": 28,
+     "patch-vars": [
+       {"name": "a",
+        "at-entry": {
+          "stored-in": "register",
+          "register": "R0"
+        },
+        "at-exit": {
+          "stored-in": "register",
+          "register": "R0"
+        }
+       },
+       {"name": "i",
+        "at-entry": {
+          "stored-in": "register",
+          "register": "R1"
+        },
+        "at-exit": {
+          "stored-in": "register",
+          "register": "R1"
+        }
+       },
+       {"name": "retval",
+        "at-entry": {
+          "stored-in": "register",
+          "register": "R3"
+        },
+        "at-exit": {
+          "stored-in": "register",
+          "register": "R3"
+        }
+       }
+     ],
+     "extra-constraints": "
```

Comment:
It seems to pick a long series of bad solutions (more than 10), so to speed things up I decided to rule out entire swaths of solutions so that it converges on the optimal one quickly.

Reply:
Not only that, but it really shouldn't be picking _any_ callee-save registers since they are not preserved in the body of the function that we're patching, thus if it picks those registers then it will break compliance with the ABI.

Reply:
Indeed, though in a perfect world this would be enforced because we include this ABI information in the property, or via the use of the patch-vars field.  extra-constraints is an "expert-level" feature that is mainly for our debugging use and not intended to be understood by end users, so I get a little squeamish when unrelated test cases depend on it.  This is fine for now, though!